moonstone/filtering/basics_filtering.py

```diff
@@ -40,5 +50,52 @@ def selecting_rows(self, desired_row_series, level_to_check):
         return self.counts_df
 
     def deleting_only_zeros_rows(self, df):
+        """DEPRECATED replaced by NoCountsFiltering"""
         self.steps.append('deleting_only_zeros_rows')
         return df[df.sum(axis=1) != 0.0]
+
+
+class NoCountsFiltering(BothAxisFiltering):
+    """
+    Remove rows (default) or columns with no counts
+    """
+
+    def filter(self) -> pd.DataFrame:
+        indices = self.df.sum(axis=self.axis) != 0.0
+        if self.axis == 1:
+            return self.df.loc[indices]
+        return self.df.loc[:, indices]
+
+
+class NamesFiltering(BothAxisFiltering):
+    """
+    Filtering based on row (default) or column names
+    """
+
+    def __init__(self, dataframe: pd.DataFrame, names: List[str], axis: int = 1, keep: bool = True):
+        """
+        :param names: list of row or column names
+        :param axis: axis to apply filtering (index (0) or columns(1))
+        :param keep: keep column (discard them if set to False)
+        """
+        self.names = names
+        self.keep = keep
+        super().__init__(dataframe, axis=axis)
+
+    def _validate_parameters(self):
+        if isinstance(self.df.index, pd.MultiIndex):
+            raise TypeError(f"{self.__class__.__name__} does not support filtering on MultiIndex dataframes.")
```

Comment:
Maybe the raised Error can suggest to use TaxonomyNamesFiltering instead

Reply:
Good idea thanks!